src/DEVICE.py

```
@@ -0,0 +1,290 @@
+			if (f != -1):
+				subprocess.call(["aplay", self.play_dir+f[0]])
+				if(DELETE == True):
+					subprocess.call(["sudo", "rm", self.play_dir+f[0]])
+			else:
+				print "No audio available."
+		
+	def record(self, COUNTDOWN, IP_ADDRESS=None):
+		self.recording_lock = threading.Lock()
+		arecord_status = self.subprocess_check_initiate("arecord", "NULL")
+		if(arecord_status == -1):
+			self.recording_event.set()
+			print "Recording stopped."
+		if(arecord_status == 0): 
+			f = time.strftime("%Y%m%d%H%M%S") + ".wav"
+			subprocess.Popen(["arecord", "-f", "dat", self.record_dir+f])
+			with self.recording_lock:
+				threading.Thread(target = self.countdown_kill, args = ("arecord", COUNTDOWN, self.recording_event)).start()
+				self.recording_event.clear()
+			print("Lock released after countdown.")
+			if(IP_ADDRESS != None):
+				try:
+					socket.inet_aton(IP_ADDRESS)
+					with self.recording_lock:
+						self.file_transfer(f, IP_ADDRESS) 
+				except socket.error:
+					raise TypeError		   
+
+class SensorTag(object):
+	
+	def __init__(self, bluetooth_adr):
+		self.con = pexpect.spawn('gatttool -b ' + bluetooth_adr + ' --interactive')
+		self.con.expect('\[LE\]>', timeout=600)
+		print "Preparing to connect. You might need to press the side button..."
+		self.con.sendline('connect')
+		self.con.expect('Connection successful.*\[LE\]>')
+		self.cb = {}
+		return
+	
+	def char_write_cmd(self, handle, value):
+		# The 0%x for value is VERY naughty!  Fix this!
+		cmd = 'char-write-cmd 0x%02x 0%x' % (handle, value)
+		print cmd
+		self.con.sendline(cmd)
+		return
+
+	def char_read_hnd(self, handle):
+		self.con.sendline('char-read-hnd 0x%02x' % handle)
+		self.con.expect('descriptor: .*? \r')
+		after = self.con.after
+		rval = after.split()[1:]
+		return [long(float.fromhex(n)) for n in rval]
+	
+	def notification_loop(self):
+		while True:
+			try:
+				pnum = self.con.expect(['Notification handle = .*? \r', "GLib-WARNING"], timeout=4)
+			except:
```

squashed exceptions are bad! http://stackoverflow.com/questions/21553327/why-is-except-pass-a-bad-programming-practice